monitoring/opencensus/main.go

```diff
@@ -0,0 +1,127 @@
+var (
+	requestCount       = stats.Int64("oc_request_count", "total request count", "requests")
+	failedRequestCount = stats.Int64("oc_failed_request_count", "count of failed requests", "requests")
+	responseLatency    = stats.Float64("oc_latency_distribution", "distribution of response latencies", "s")
+)
+
+// [END monitoring_sli_metrics_opencensus_measure]
+
+// set up views
+// [START monitoring_sli_metrics_opencensus_view]
+var (
+	requestCountView = &view.View{
+		Name:        "oc_request_count",
+		Measure:     requestCount,
+		Description: "total request count",
+		Aggregation: view.Count(),
+	}
+	failedRequestCountView = &view.View{
+		Name:        "oc_failed_request_count",
+		Measure:     failedRequestCount,
+		Description: "count of failed requests",
+		Aggregation: view.Count(),
+	}
+	responseLatencyView = &view.View{
+		Name:        "oc_response_latency",
+		Measure:     responseLatency,
+		Description: "The distribution of the latencies",
+		// bucket definitions must be explicit
+		Aggregation: view.Distribution(0, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000),
+	}
+)
+
+// [END monitoring_sli_metrics_opencensus_view]
+
+func main() {
+
+	// set up project ID
+	projectID := flag.String("project_id", "", "Cloud Project ID")
+	flag.Parse()
+
+	// register the views
+	if err := view.Register(requestCountView, failedRequestCountView, responseLatencyView); err != nil {
+		log.Fatalf("Failed to register the views: %v", err)
+	}
+	// [START monitoring_sli_metrics_opencensus_exporter]
+	// set up Cloud Monitoring exporter
+	sd, err := stackdriver.NewExporter(stackdriver.Options{
+		ProjectID:         *projectID,
+		MetricPrefix:      "opencensus-demo",
+		ReportingInterval: 60 * time.Second,
+	})
+	if err != nil {
+		log.Fatalf("Failed to create the Cloud Monitoring exporter: %v", err)
+	}
+	defer sd.Flush()
+	// Start the metrics exporter
+	sd.StartMetricsExporter()
+	defer sd.StopMetricsExporter()
+	// [END monitoring_sli_metrics_opencensus_exporter]
+	http.HandleFunc("/", handle)
+	log.Fatal(http.ListenAndServe(":8080", nil))
+}
+
+func handle(w http.ResponseWriter, r *http.Request) {
+	// create context
+	ctx, _ := tag.New(context.Background())
+	requestReceived := time.Now()
+	// [START monitoring_sli_metrics_opencensus_counts]
+	// count the request
+	stats.Record(ctx, requestCount.M(1))
+
+	// fail 10% of the time
+	if rand.Intn(100) > 90 {
+		// count the failed request
+		stats.Record(ctx, failedRequestCount.M(1))
+		// [END monitoring_sli_metrics_opencensus_counts]
+		fmt.Fprintf(w, "intentional error!")
+		// record latency for failure
+		// [START monitoring_sli_metrics_opencensus_latency]
+		stats.Record(ctx, responseLatency.M(time.Since(requestReceived).Seconds()))
+		// [END monitoring_sli_metrics_opencensus_latency]
+		return
+	} else {
+		delay := time.Duration(rand.Intn(1000)) * time.Millisecond
+		time.Sleep(delay)
+		fmt.Fprintf(w, "Succeeded after "+strconv.Itoa(int(delay)/1000000)+" ms")
```

Comment:
`fmt.Fprintf` can print using C-style placeholders. You should be able to do:

```suggestion
		fmt.Fprintf(w, "Succeeded after %v", delay)
```

See https://play.golang.org/p/QHPpWNOZqz2.

Reply:
Fixed, thank you!